rflx/expression.py

```diff
@@ -81,6 +81,9 @@ def __init__(
         self.type_ = type_
         self.location = location
         self.error = error or RecordFluxError()
+        self._z3expr: Optional[z3.ExprRef] = None
+        self._z3arithexpr: Optional[z3.ArithRef] = None
+        self._z3boolexpr: Optional[z3.BoolRef] = None
```

Comment:
Is there a reason for implementing the caching manually instead of using [`@lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache) for the `z3expr` method?

Reply:
Didn't think about this. Surprisingly the LRU cache is 5% faster than the manual implementation and 50% faster than the uncached variant (tested using the prepare step of rfzz on IPv4 with IHL limited to 6). Changed in 83083cf.